### PR TITLE
fix(chore): Handle spaces in git command arguments

### PR DESCRIPTION
**Pull request by rashil2000**
#### Description

Fix regression introduced in #5122, i.e. handling of Scoop home paths which have spaces in them.

#### How Has This Been Tested?

Run commands like `scoop bucket ...` or `scoop info ...` which use Git internally.

#### Checklist:

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).
- [x] I have ensured that I am targeting the `develop` branch.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.
- [x] I have added an entry in the CHANGELOG.

### CODE REVIEW

**Comment by rashil2000** [comment]
@niheaven could you please review? This is a crucial fix for me as I have a non-standard Scoop home folder.

File: lib/core.ps1 [hunk 1]
```diff
@@ -145,19 +145,16 @@ function Invoke-Git {
...
 
     if ($WorkingDirectory) {
-        $cmd = "`"$git`" -C `"$WorkingDirectory`" $arguments"
+        $ArgumentList = @('-C', $WorkingDirectory) + $ArgumentList
     }
-    $sb = [scriptblock]::Create("& $cmd")
 
     if([String]::IsNullOrEmpty($proxy) -or $proxy -eq 'none')  {
-        return Invoke-Command $sb
+        return & $git $ArgumentList
```
Comment (niheaven) [inline lib/core.ps1:154]:
```suggestion
        return & $git $ArgumentList
```

This should be `$ArgumentList` or `@ArgumentList`? I mean splatting.

Comment (rashil2000) [reply lib/core.ps1:154]: Isn't splatting used for hashtables? `$ArgumentList` here is just a string array. Nevertheless, I tried a few complex commands and `$ArgumentList` seems to work fine.

Comment (niheaven) [reply lib/core.ps1:154]: https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_splatting?view=powershell-5.1

I think you should use splatting here for clear command.

Comment (rashil2000) [reply lib/core.ps1]: Done

**Review by niheaven — APPROVED** [review]
LGTM

**Pull request merged**